src/mono/mono/component/component.h

```diff
@@ -0,0 +1,16 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+
+#ifndef _MONO_COMPONENT_COMPONENT_H
+#define _MONO_COMPONENT_COMPONENT_H
+
+typedef struct _MonoComponent MonoComponent;
+
+typedef void (*MonoComponent_CleanupFn) (MonoComponent *self);
+
+struct _MonoComponent {
+	MonoComponent_CleanupFn cleanup;
```

Comment:
next PR we should think about whether this makes sense since we don't really do cleanup on shutdown anymore.

It might be good to put in something like
```c
typedef int (*MonoComponent_AvailableFn) (void);

struct _MonoComponent {
    intptr_t itf_version;
    MonoComponent_AvailableFn available;
};
```

We could require `itf_version` to be some magic number `MONO_COMPONENT_ITF_VERSION` and check at load time to make sure there's no packaging mistakes.

After that, calling `mono_component_event_pipe()->available()` will return 0 from the stub and non-zero from the real component.

Reply:
Since I address other review feedback in this PR I did this change as well. Dropped cleanup and added an available that can be used by callers if needed. Used a bool instead of an int as return value. Added the itf_version field into component struct as well and added code into init checking that it matches the version expected by the runtime, if not we will assert at startup.